sql-plugin/src/main/scala/com/nvidia/spark/rapids/HostColumnarToGpu.scala

```diff
@@ -16,16 +16,73 @@
 
 package com.nvidia.spark.rapids
 
+import java.nio.ByteBuffer
+
+import org.apache.arrow.vector.ValueVector
+
 import org.apache.spark.TaskContext
+import org.apache.spark.internal.Logging
 import org.apache.spark.rdd.RDD
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.catalyst.expressions.Attribute
 import org.apache.spark.sql.execution.{SparkPlan, UnaryExecNode}
 import org.apache.spark.sql.execution.metric.{SQLMetric, SQLMetrics}
 import org.apache.spark.sql.types._
-import org.apache.spark.sql.vectorized.{ColumnarBatch, ColumnVector}
+import org.apache.spark.sql.vectorized.{ArrowColumnVector, ColumnarBatch, ColumnVector}
+import org.apache.spark.sql.vectorized.rapids.AccessibleArrowColumnVector
+
+object HostColumnarToGpu extends Logging {
+
+  // use reflection to get value vector from ArrowColumnVector
+  private def getArrowValueVector(cv: ColumnVector): ValueVector = {
+    val arrowCV = cv.asInstanceOf[ArrowColumnVector]
+    val fields = arrowCV.getClass.getDeclaredFields.toList
+    val field = fields.filter( x => {
+      x.getName.contains("accessor")
+    }).head
+    field.setAccessible(true)
+    val accessor = field.get(arrowCV)
+    val accessFields = accessor.getClass().getSuperclass().getDeclaredFields().toList
+    val valVecField = accessFields.filter( x => {
+      x.getName.contains("vector")
+    }).head
+    valVecField.setAccessible(true)
+    valVecField.get(accessor).asInstanceOf[ValueVector]
+  }
+
+  def arrowColumnarCopy(
+      cv: ColumnVector,
+      ab: ai.rapids.cudf.ArrowColumnBuilder,
+      nullable: Boolean,
+      rows: Int): Unit = {
+    val valVector = if (cv.isInstanceOf[ArrowColumnVector]) {
```

Comment:
nit: Why not use a match instead of is/else?  It feels more scala like to do it that way.